File: app/src/main/java/com/example/olesya/boardgames/interfaces/ImageHolderDao.java
package com.example.olesya.boardgames.interfaces;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.example.olesya.boardgames.models.ImageHolder;

import java.util.ArrayList;
import java.util.List;

@Dao
public interface  ImageHolderDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(ImageHolder... imgs);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(ArrayList<ImageHolder> imgs);

    // Удаление Person из бд
    @Delete
    void delete(ImageHolder img);

    // Получение всех Person из бд
    @Query("SELECT * FROM imageholder")
    List<ImageHolder> getAllImages();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(ImageHolder holder);
}

File: app/src/main/java/com/example/olesya/boardgames/database/ImageHolderDao.kt
package com.example.olesya.boardgames.database

import androidx.room.*
import com.example.olesya.boardgames.entity.ImageHolder

@Dao
interface ImageHolderDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(imgs: List<ImageHolder>)

    @Delete
    fun delete(img: ImageHolder)

    @Query("SELECT * FROM imageholder")
    fun getAllImages(): List<ImageHolder>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(holder: ImageHolder)
}
